mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx

```diff
@@ -56,9 +56,14 @@ export const renderModalContent = (params: RenderModalContentParams) => {
 /**
  * Get user name
  */
-const getName = (mentors: Map<string, UserPreview>, mentorUuid: string, ownerName: string) => {
-  const mentor = mentors.get(mentorUuid);
-  const name = mentor ? mentor.name : ownerName;
+const getName = (
+  mentors: Map<string, UserPreview>,
+  formerMentors: Map<string, UserPreview>,
+  mentorUuid: string,
+  ownerName: string) => {
+  const mentorName = mentors.get(mentorUuid) || formerMentors.get(mentorUuid);
+
+  const name = mentorName ? mentorName.name : ownerName;
```

Comment:
lets improve logice genName, for now it spreaded across few lines. I propose to put all users in some structure like map and use it to access appropriate name